src/VideoComponents/MiniMapWidget.jsx

```diff
@@ -0,0 +1,220 @@
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import './css/MiniMapWidget.css';
+import { mapRange } from './js/tools.js';
+
+function drawBoat(ctx, boatWidth, boatLength) {
+  ctx.beginPath();
+  ctx.moveTo(0, -boatLength / 2);
+  ctx.lineTo(boatWidth, -boatWidth);
+  ctx.lineTo(boatWidth, boatLength / 2);
+  ctx.lineTo(-boatWidth, boatLength / 2);
+  ctx.lineTo(-boatWidth, -boatWidth);
+  ctx.closePath();
+  ctx.fillStyle = '#FFFFFF';
+  ctx.fill();
+}
+
+function drawROV(ctx, rovSize) {
+  ctx.beginPath();
+  ctx.moveTo(-rovSize / 2, -rovSize / 2);
+  ctx.lineTo(rovSize / 2, -rovSize / 2);
+  ctx.lineTo(rovSize / 2, rovSize / 2);
+  ctx.lineTo(-rovSize / 2, rovSize / 2);
+  ctx.lineTo(-rovSize / 2, -rovSize / 2);
+  ctx.moveTo(0, -rovSize / 2);
+  ctx.lineTo(rovSize / 3, -rovSize / 3 - rovSize / 2);
+  ctx.lineTo(-rovSize / 3, -rovSize / 3 - rovSize / 2);
+  ctx.closePath();
+}
+
+function drawNEDframe(ctx, mapWidth, boatWidth, rovSize) {
+  ctx.beginPath();
+  //east arrow
+  ctx.moveTo(0, 0);
+  ctx.lineTo(mapWidth / 2 - boatWidth - rovSize / 2, 0);
+  ctx.lineTo(
+    mapWidth / 2 - boatWidth - rovSize / 2 - mapWidth / 40,
+    mapWidth / 40,
+  );
+  ctx.moveTo(mapWidth / 2 - boatWidth - rovSize / 2, 0);
+  ctx.lineTo(
+    mapWidth / 2 - boatWidth - rovSize / 2 - mapWidth / 40,
+    -mapWidth / 40,
+  );
+  //nort arrow
+  ctx.moveTo(0, 0);
+  ctx.lineTo(0, -(mapWidth / 2 - boatWidth - rovSize / 2));
+  ctx.lineTo(
+    -mapWidth / 40,
+    -(mapWidth / 2 - boatWidth - rovSize / 2) + mapWidth / 40,
+  );
+  ctx.moveTo(0, -(mapWidth / 2 - boatWidth - rovSize / 2));
+  ctx.lineTo(
+    mapWidth / 40,
+    -(mapWidth / 2 - boatWidth - rovSize / 2) + mapWidth / 40,
+  );
+  ctx.stroke();
+
+  ctx.strokeText(
+    'N',
+    mapWidth / 40,
+    -(mapWidth / 2 - boatWidth - rovSize / 2) + mapWidth / 40,
+  );
+  ctx.strokeText(
+    'E',
+    mapWidth / 2 - boatWidth - rovSize - mapWidth / 40,
+    mapWidth / 15,
+  );
+}
+
+function drawArrow(ctx, rovSize, mapWidth) {
+  ctx.beginPath();
+  ctx.moveTo(mapWidth / 2, 0);
+  ctx.lineTo(mapWidth / 2 - rovSize, rovSize);
+  ctx.lineTo(mapWidth / 2 - rovSize, -rovSize);
+  ctx.closePath();
+}
```

Comment:
Maybe extract these to separate .js-file, like Aleksander has done in the other widgets using canvas?